homeassistant/util/package.py

```diff
@@ -8,9 +8,19 @@
 from typing import Optional
 from urllib.parse import urlparse
 
-from importlib_metadata import PackageNotFoundError, version
 import pkg_resources
 
+if sys.version_info[:2] >= (3, 8):
```

Comment:
We can test this by patching `sys.version_info`:
https://github.com/home-assistant/core/blob/4412ce3b86e886292ad662dc483830af37c2b4fd/tests/test_main.py#L29

Reply:
(Apologies if these are silly questions, I'm not a real Python person :joy:)

Does it make sense to patch `sys.version_info`?
* If the Python37 test patches `sys.version_info` to 3.8, then the (first) import will fail because `importlib.metadata` is only available on 3.8 and above
* If the Python38 test patches `system.version_info` to 3.7, then the (second) import will also fail because the `importlib_metadata` dependency is only installed for 3.7 and below

Or do you mean there should be an import test where failure is the intended result?

Also, I would have thought that the important part of this overall change is that `package.is_installed()` still functions correctly in 3.8 and newer? If so, then `test_check_package_global()` and `test_check_package_zip()` (in tests/util/test_package.py) would continue to test this functionality?

Reply:
I think it's ok to not add a unit test, since we test this module for both Python 3.7 and 3.8.